test/test_lswt.jl

```diff
@@ -0,0 +1,109 @@
+@testitem "Single Ion" begin
+    J, J′, D = 1.0, 0.1, 5.0
+
+    a = b = 1.0
+    c = 1.5
+    lat_vecs = lattice_vectors(a, b, c, 90, 90, 90)
+    types = ["A"]
+    basis_vecs = [[0, 0, 0]]
+
+    cryst = Crystal(lat_vecs, basis_vecs; types)
+
+    # Spin System
+    dims = (2, 2, 2)
+    infos = [SpinInfo(1, S=1)]
+    sys = System(cryst, dims, infos, :SUN)
+
+    set_exchange!(sys, J,  Bond(1, 1, [1, 0, 0]))
+    set_exchange!(sys, J′, Bond(1, 1, [0, 0, 1]))
+    set_anisotropy!(sys, D * 𝒮[3]^2, 1)
+
+    Δt  = abs(0.05 / D)
+    λ = 0.1
+    langevin = Langevin(Δt; kT=0, λ)
+
+    randomize_spins!(sys)
+    A = [1 1 1; -1 1 0; 0 0 1]
+    sys_lswt = construct_magnetic_supercell(sys, [1 1 1; -1 1 0; 0 0 1])
+
+    langevin.kT = 0
+    for i in 1:50_000
+        step!(sys_lswt, langevin)
+    end
+
+    sw_fields = SpinWaveFields(sys_lswt)
+
+    function sion_analytical_disp(k :: Vector{Float64})
+        # analytical solutions
+        γkxy = cos(2*π*k[1]) + cos(2*π*k[2])
+        γkz  = cos(2*π*k[3])
+        x = 1/2 - D/(8*(2*J+J′))
+        Ak₊ = -8 * (x-1) * x * (2*J+J′) - (x-1) * D + 2 * (2*x-1) * (J *γkxy + J′*γkz)
+        Bk₊ = -2 * (J * γkxy + J′ * γkz)
+        Ak₋ = -16 * (x-1) * x * (2*J+J′) - (2*x-1) * D - 2 * (1-2*x)^2*(J*γkxy + J′*γkz)
+        Bk₋ = 2 * (1-2*x)^2 * (J*γkxy + J′*γkz)
+        ωk₊ = √(Ak₊^2-Bk₊^2)
+        ωk₋ = √(Ak₋^2-Bk₋^2)
+        return ωk₊, ωk₋
+    end
+
+    k = rand(Float64, 3)
+    ωk1, ωk2 = sion_analytical_disp(k)
+    ωk3, ωk4 = sion_analytical_disp(k .+= 0.5)
+    ωk_ana = [ωk1, ωk2, ωk3, ωk4]
+    index  = sortperm(ωk_ana, rev=true)
+    ωk_ana = ωk_ana[index]
+    ωk_num = lswt_dispersion_relation(sw_fields, k)
+
+    @test isapprox(ωk_ana, ωk_num)
+end
+
+@testitem "Intensities" begin
+    using LinearAlgebra
+
+    a = 8.289
+    lat_vecs = lattice_vectors(a, a, a, 90, 90, 90)
+    types = ["MzR1"]
+    basis_vecs = [[0, 0, 0]]
+    fcc = Crystal(lat_vecs, basis_vecs, 225; types)
+    S = 5/2
+
+    # According to a renormalized classical theory for spins (the details will be presented in a manuscript in preparation), the large-S expansion and the :dipole mode should produce the same results when
+    cov_factor = (1 - 3/S + 11/(4*S^2)- 3/(4*S^3))
+
+    dims = (1, 1, 1)
+    infos = [SpinInfo(1, S=S)]
+    sys = System(fcc, dims, infos, :dipole)
+
+    J = 22.06 * Sunny.meV_per_K
+    K = 0.15  * Sunny.meV_per_K
+    C = J + K
+    J₁ = diagm([J, J, C])
+    D_ST = 0.2
+    D = D_ST / cov_factor
+
+    set_exchange!(sys, J₁, Bond(1, 2, [0, 0, 0]))
+    Λ = D * (𝒮[1]^4 + 𝒮[2]^4 + 𝒮[3]^4)
+    set_anisotropy!(sys, Λ, 1)
+
+    Δt = abs(0.05 / D)
+    λ  = 0.1
+    langevin = Langevin(Δt; kT=0, λ)
+
+    dipoles = S * [1.0 1.0 -1.0 -1.0; 1.0 -1.0 -1.0 1.0; 1.0 -1.0 1.0 -1.0] / √3
+    positions = [0.0 0.5 0.5 0.0; 0.0 0.5 0.0 0.5; 0.0 0.0 0.5 0.5]
+    set_dipoles!(sys, dipoles, positions)
```

Comment:
Are you ok with replacing this as:
```
polarize_spin!(sys, (1, 1, 1), position_to_site(sys, (0, 0, 0)))
polarize_spin!(sys, (1, -1, -1), position_to_site(sys, (1/2, 1/2, 0)))
...
```

Instead of `polarize_spin!` there is also `set_coherent_state!` which functions similarly.

Reply:
`polarize_spin!` looks cleaner than `set_dipoles`. I have updated these lines and removed `set_dipoles`.